[#77] Labels API 기본 CRUD 구현

junsushin-dev opened the pull request:
## 구현의도
- Label에 대한 기본적인 CRUD 작업을 하는 API 입니다. 

## 흐름도, 다이어그램
- 작업흐름
  - routes/labels -> services/labelServices -> db/models -> DB 서버

## 사용된 기술
- sequelize

## 리뷰요청 & 논의사항 & 궁금한점
- PUT, DELETE 문의 구현 방식과 DB에서 변경이 잘 이루어졌는지 확인하는 방법을 비교하고 논의해보고 싶습니다. 

closes #77

junsushin-dev committed:
feat: labels API 기본 CRUD 구현

update, delete 시에는 변경된 row 개수 체크

passwd10 commented on backend/src/routes/labels.js:
```diff
@@ -1,21 +1,42 @@
 const express = require('express');
 const router = express.Router();
+const createError = require('http-errors');
+const { getLabels, addLabel, updateLabel, deleteLabel } = require('../services/labelService');
```
import 하는곳의 역할이 달라서 개행으로 구분해주시면 좋을것같습니다! :smile:

junsushin-dev replied:
라이브러리 단위 import와 프로젝트 내 타 모듈 import는 구분해 주는 것이 저도 좋다고 생각합니다. 좋은 지적 감사드려요! 😁

passwd10 commented on backend/src/routes/labels.js:
```diff
@@ -1,21 +1,42 @@
+router.post('/', async (req, res, next) => {
+  try {
+    await addLabel(req.body);
+    res.sendStatus(201);
```
wiki API 명세에 post, put, delete 성공시 success message를 보내는것으로 적혀있네요!

junsushin-dev replied:
헉 다시 Wiki를 보니까 그렇네요... 명세를 이미 다 알고 있다고 생각했는데, 역시 보면서 해야 하는 것 같네요. 수정하도록 하겠습니다. 🥩

passwd10 commented on backend/src/services/labelService.js:
```diff
@@ -0,0 +1,39 @@
+const getLabels = async () => {
+  const labels = await Label.findAll({
+    attributes: {
+      exclude: ['isDeleted', 'createdAt', 'updatedAt'],
```
이렇게하면 불필요한 데이터를 받지 않을 수 있군요! 제 코드에도 적용해보겠습니다 :+1:

junsushin-dev replied:
감사합니다! 😀

passwd10 commented on backend/src/services/labelService.js:
```diff
@@ -0,0 +1,39 @@
+};
+
+const updateLabel = async (label) => {
+  const { id, ...rest } = label;
```
구조분해할당을 이렇게도 사용할 수 있군요 좋네요! :+1:

junsushin-dev replied:
감사합니다! 논외로 label, rest가 좋은 변수명인지는 조금 의문이었는데, 인서님 코드를 보니까 변수명을 잘 지으신듯 해서 변수 이름을 바꾸어보려고 합니다. ☺

passwd10 reviewed (commented):
준수님 리뷰 검토부탁드립니다!

junsushin-dev commented:
회고하다가 떠올랐는데, 아직 labels의 PUT, DELETE 요청 key를 title로 바꾸어놓지 않아서 변경하고 다시 오도록 하겠습니다.

junsushin-dev committed:
refactor: PR #90 리뷰사항 반영

import문 사이 개행 추가해서 라이브러리와 프로젝트 내 import 구분
API 명세에 따라 POST, PUT, DELETE response에 메시지 추가
가독성 향상을 위한 변수 할당 및 변수 리팩토링
DELETE 문에는 id만 입력변수로 필요하기 때문에 변경

junsushin-dev commented:
@passwd10 인서님 리뷰사항 반영 완료하였습니다! 😋

passwd10 commented:
> @passwd10 인서님 리뷰사항 반영 완료하였습니다!

고생하셨습니다 ㅎㅎ @parkdit94 동현님께서 리뷰해주시고 MERGE 해주시면 될것같아요!

dhyun2 reviewed (approved):
수고많으셨습니다!! 😊
